server/util/db/db.go

```diff
@@ -523,6 +528,19 @@ func ParseDatasource(fileResolver fs.FS, datasource string, advancedConfig *Adva
 			return nil, fmt.Errorf("malformed db connection string")
 		}
 		driverName, connString := parts[0], parts[1]
+		switch driverName {
```

Comment:
This could obviously be an `if` right now, but the postgres driver has different logic here so it eventually is a `switch` anyway, so I just left it.